components/suggest/src/db.rs

```diff
@@ -123,14 +124,29 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given keyword from the database.
     pub fn fetch_by_keyword(&self, keyword: &str) -> Result<Vec<Suggestion>> {
+        let pocket_keyword = match keyword.contains(' ') {
+            true => keyword.to_string(),
+            false => keyword.to_owned() + " ",
+        };
         self.conn.query_rows_and_then_cached(
-            "SELECT s.id, k.rank, s.title, s.url, s.provider
+            "SELECT s.id, k.rank, s.title, s.url, s.provider, false as is_top_pick
              FROM suggestions s
              JOIN keywords k ON k.suggestion_id = s.id
              WHERE k.keyword = :keyword
+             UNION
+             SELECT s.id, k.rank, s.title, s.url, s.provider, true as is_top_pick
+             FROM suggestions s
+             JOIN pocket_high_confidence_keywords k ON k.suggestion_id = s.id
+             WHERE k.keyword = :keyword
+             UNION
+             SELECT s.id, k.rank, s.title, s.url, s.provider, false as is_top_pick
+             FROM suggestions s
+             JOIN pocket_low_confidence_keywords k ON k.suggestion_id = s.id
+             WHERE k.keyword BETWEEN :pocket_keyword AND :pocket_keyword || x'ffff'
              LIMIT 1",
```

Comment:
Now that we have multiple keyword sources, maybe it's time to return all the hits back (sorted by score) to the consumer instead of 1. Even returning the one with the highest score could lead to some surprises, e.g. if the user dismissed that suggestion, all other candidates for the same keyword will never get a chance to show.

Also cc @linabutler @0c0w3 for their thoughts.